### ISSUE TITLE
Pull queries do not update if their nested entities change

### ISSUE THREAD

**Issue by seantempesta** [mpdairy/posh#26]
Ran into this bug today.  I had a `(pull conn '[*] eid)` that wasn't updating when a nested entity within that pull changed.  I'm guessing it's because the datom match pattern is just for the specified eid (and not all returned entities)?

Anyway, I setup an example showing the bug here:
https://github.com/seantempesta/posh-bug

The jist of it is this:
```clojure
(def schema {;; Contact component (shared across entities)
             :contact/email {:db/cardinality :db.cardinality/one}
             :contact/phone {:db/cardinality :db.cardinality/one}

             ;; Users
             :user/username {:db/unique      :db.unique/identity
                             :db/cardinality :db.cardinality/one}
             :user/contact  {:db/valueType   :db.type/ref
                             :db/cardinality :db.cardinality/one
                             :db/isComponent true}})
(def conn (d/create-conn schema))
(posh! conn)


(transact! conn [{:db/id         -1
                  :user/username "seantempesta"
                  :user/contact  {:db/id         -2
                                  :contact/phone "555-5555"
                                  :contact/email "sean.tempesta@gmail.com"}}])



;; -------------------------
;; Views

(defn home-page []
  (let [user (pull conn '[*] [:user/username "seantempesta"])
        phone (pull conn '[*] 2)]
    [:div [:h2 "Welcome to posh-bug!"]

     [:div [:h4 {} "User Entity"]
      [:h5 {} @user]]

     [:div [:h4 {} "Phone Entity"]
      [:h5 {} @phone]]

     [:input {:type     "button" :value "Remove phone number"
              :on-click #(transact! conn [[:db/retract 2 :contact/phone "555-5555"]])}]]))
```
<img width="599" alt="screen shot 2017-04-30 at 11 44 37 am" src="https://cloud.githubusercontent.com/assets/726276/25561661/33c034ee-2d9b-11e7-9c58-6264c8a2a4b0.png">

**Comment by seantempesta** [comment]
So, I'm really not familiar enough with Posh's internals to fix this.  I thought I saw somewhere that Posh might bring back manual specification of listen patterns?   At least then I could work around this bug.

**Comment by seantempesta** [comment]
Temporary workaround.  I found if you explicitly expand isComponent pull patterns the pull analyze picks up the changes.  

So, in the example above, you'd have to do something like this:
```clojure
(pull conn '[:db/id, :user/username, {:user/contact [:db/id, :contact/email, :contact/phone]}] 2)
```

**Comment by metasoarous** [comment]
Yes; it should certainly be looking at isComponent attributes. I'm flagging as bug. Thanks for reporting.

**Comment by eoliphan** [comment]
@seantempesta was that all you had to do?  I've tried the explicit expansion, still no joy for my reactions

**Comment by seantempesta** [comment]
@eoliphan: It worked for my scenario at that time.  No idea if it's working now as I'm no longer using Posh.

**Comment by eoliphan** [comment]
@seantempesta ok, thanks. Yeah, i'm looking at alternatives as well, I really wanted to use DS as I've got Datomic on the server, but probably will just go with specter and maps, etc. for now until I can maybe look at figuring out the posh code and doing some hacking.

**Comment by tkachenko1503** [comment]
Hello guys! I've got the same issue with nested components. Are there some ways to solve this issue? I can try to help with it if someone has an idea how to fix this bug.

**Comment by fahtative** [comment]
I wrote this quick fix

**Comment by fahtative** [comment]
I wrote this fix: `[gist](https://gist.github.com/fahtative/a23474e7833ee4f17d1b92534a494400)`.
It's working, but I have no idea about efficiency...

**Comment by fahtative** [comment]
I wrote this [fix](https://gist.github.com/fahtative/a23474e7833ee4f17d1b92534a494400) if anyone's interested. It basically express a `pull` as a `query`.
